tables/TaQL/test/tExprNodeSetOpt.cc

```diff
@@ -0,0 +1,361 @@
+//# tExprNodeSetOpt.cc: Test program for the ExprNodeSeOpt classes
+//# Copyright (C) 2022
+//# Associated Universities, Inc. Washington DC, USA.
+//#
+//# This program is free software; you can redistribute it and/or modify it
+//# under the terms of the GNU General Public License as published by the Free
+//# Software Foundation; either version 2 of the License, or (at your option)
+//# any later version.
+//#
+//# This program is distributed in the hope that it will be useful, but WITHOUT
+//# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+//# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+//# more details.
+//#
+//# You should have received a copy of the GNU General Public License along
+//# with this program; if not, write to the Free Software Foundation, Inc.,
+//# 675 Massachusetts Ave, Cambridge, MA 02139, USA.
+//#
+//# Correspondence concerning AIPS++ should be addressed as follows:
+//#        Internet email: aips2-request@nrao.edu.
+//#        Postal address: AIPS++ Project Office
+//#                        National Radio Astronomy Observatory
+//#                        520 Edgemont Road
+//#                        Charlottesville, VA 22903-2475 USA
+
+#include <casacore/tables/TaQL/ExprNodeSetOpt.h>
+#include <casacore/tables/TaQL/ExprNodeSet.h>
+#include <casacore/tables/TaQL/ExprNode.h>
+#include <casacore/casa/Arrays/ArrayLogical.h>
+#include <casacore/casa/Utilities/Assert.h>
+#include <casacore/casa/iostream.h>
+
+#include <casacore/casa/namespace.h>
+
+// <summary>
+// Test program for the ExprNodeSetOpt classes.
+// </summary>
+
+
+
+void doDoubleContSet()
+{
+  std::vector<double> st ({1,3,5,7,9,11,13,15,17,19,21});
```

Comment:
Please test with vectors with different sizes too.

Reply:
Overhauled the test program. Using templated function to test find and contains.
Expected result is now a Vector making it more clear.
This also addresses the next points.